### DISCUSSION TITLE
New to Gorgon

### DISCUSSION THREAD

**Discussion by PaulDirectX** [Tape-Worm/Gorgon#42] [Q&A] [answered]
Hi,
I've worked with DX in University (9 years ago...) and I know are a lot to init to a DX app. 
I wanted to give a shot to SharpDX 3 weeks ago and find your Gorgon. 
Question is , is there any starting point project, from where you can build code with already included gorgon and all buffers (back / pixelshader) in place?  Or a step by step function calls in order to build a project and actual create primitives in the end. 
Like a Simple box scene project with gorgon where I can rotate and pan will do to understand the project structure. Nothing fancy or complex optimization. 

For the moment I've just build a project that swaps backbuffer with a simple gradient from light blue to blue with SharpDX.  
Any links / guides that can give me an overview of function calls and order of calling them will do also. 

Many thanks

**Comment by Tape-Worm** [comment, marked as answer]
Yep, there's a full suite of example applications included that show how to build a simple program to larger, more complex applications.

These are all included in the project solution under the Examples solution folder. 
<img width="197" alt="image" src="https://user-images.githubusercontent.com/9710137/149569656-df26506c-5884-48ae-9481-e2df68662209.png">

With your current gradient fill application, you could do it like this with Gorgon:
```csharp
GorgonGraphics gfx;
GorgonSwapChain swap;
Gorgon2D renderer;
GorgonSprite sprite;

void Main()
{
  Application.SetHighDpiMode(HighDpiMode.PerMonitorV2);
  Application.EnableVisualStyles();
  Application.SetCompatibleTextRenderingDefault(false);

  try
  {
      // Now begin running the application idle loop.
      GorgonApplication.Run(Init(), Idle);
  }
  catch (Exception ex)
  {
      GorgonExample.HandleException(ex);
  }
  finally
  {
      renderer?.Dispose();
      swap?.Dispose();
      gfx?.Dispose();
  }  
}

void Init()
{
  IReadOnlyList<IGorgonVideoAdapterInfo> devices = GorgonGraphics.EnumerateAdapters();
  
  // Now we create the main graphics interface with the first applicable video device.
  gfx = new GorgonGraphics(devices[0]);
  swap = new GorgonSwapChain(gfx, yourWindow, new GorgonSwapChainInfo(yourWindow.ClientSize.Width,
                                                                           yourWindow.ClientSize.Height,
                                                                           BufferFormat.R8G8B8A8_UNorm));
  renderer = new Gorgon2D(swap);
  sprite = new GorgonSprite
  {
   Width = swap.Width,
   Height = swap.Height,
   CornerColors =
   {
      UpperLeft = GorgonColor.LightBlue,
      UpperRight = GorgonColor.LightBlue,
       LowerLeft = GorgonColor.BluePure,
       LowerRight = GorgonColor.BluePure,
   }
  };

  gfx.SetRenderTarget(swap.RenderTargetView);
}

bool Idle()
{
  swap.RenderTargetView.Clear(GorgonColor.Black);

  renderer.Begin();
  renderer.DrawSprite(sprite);
  renderer.End();

  swap.Present(1);
}
```
I give no guarantees that the above code will actually compile/run. But that's roughly what you'll need to do to replicate your SharpDX example. And the total code footprint should be much smaller too (I hope so anyway, that's the point of this).

**Reply by PaulDirectX** [reply]
Thank you for interest to solve this. 
I'll give it a try now.

**Reply by Tape-Worm** [reply]
Also, for future reference:
https://www.tape-worm.net/GorgonDocs/articles/GettingStarted.html
